### ISSUE TITLE
Generated ids are invalid when using [GenerateSerializer(GenerateFieldIds = GenerateFieldIds.PublicProperties)]

### ISSUE THREAD

**Issue by zpxp** [dotnet/orleans#8245]
When using the `GenerateFieldIds.PublicProperties`  in `[GenerateSerializer]` to auto generate the ids, some of them are invalid. Inside the gnerated codec `Deserialize` function, the hard coded ids are negative, and when compared with a `uint`, the if statement will never evaluate to true, meaning that member will never deserialize. Here is an example:

```cs
  [global::System.Runtime.CompilerServices.MethodImplAttribute(global::System.Runtime.CompilerServices.MethodImplOptions.AggressiveInlining)]
        public void Deserialize<TReaderInput>(ref global::Orleans.Serialization.Buffers.Reader<TReaderInput> reader, global::Game.Core.ClickPipeModel instance)
        {
            uint id = 0;
            global::Orleans.Serialization.WireProtocol.Field header = default;
            while (true)
            {
                reader.ReadFieldHeader(ref header);
                if (header.IsEndBaseOrEndObject)
                    break;
                id += header.FieldIdDelta;
                if (id == 1784858566)
                {
                    instance.HighlightColor = (global::Game.Core.ClickPipeHighlightModel)_codec1.ReadValue(ref reader, header);
                    reader.ReadFieldHeader(ref header);
                    if (header.IsEndBaseOrEndObject)
                        break;
                    id += header.FieldIdDelta;
                }

                if (id == -1917727542)
                {
                    instance.Staged = (global::System.Collections.Generic.List<global::OneOf.OneOf<global::Game.Core.GameCardModel, global::Game.Core.GamePlayerModel>>)_codec0.ReadValue(ref reader, header);
                    reader.ReadFieldHeader(ref header);
                    if (header.IsEndBaseOrEndObject)
                        break;
                    id += header.FieldIdDelta;
                }

                if (id == -1107141278)
                {
                    instance.CanStage = (global::System.Collections.Generic.List<global::OneOf.OneOf<global::Game.Core.GameCardModel, global::Game.Core.GamePlayerModel>>)_codec0.ReadValue(ref reader, header);
                    reader.ReadFieldHeader(ref header);
                    if (header.IsEndBaseOrEndObject)
                        break;
                    id += header.FieldIdDelta;
                }

                if (id == -627159035)
                {
                    instance.Text = (string)global::Orleans.Serialization.Codecs.StringCodec.ReadValue(ref reader, header);
                    reader.ReadFieldHeader(ref header);
                    if (header.IsEndBaseOrEndObject)
                        break;
                    id++;
                }

                reader.ConsumeUnknownField(ref header);
            }
        }
```

When i manually specify the `[Id(1)]` per property, this is resolved

```cs
  [global::System.Runtime.CompilerServices.MethodImplAttribute(global::System.Runtime.CompilerServices.MethodImplOptions.AggressiveInlining)]
        public void Deserialize<TReaderInput>(ref global::Orleans.Serialization.Buffers.Reader<TReaderInput> reader, global::Game.Core.ClickPipeModel instance)
        {
            uint id = 0;
            global::Orleans.Serialization.WireProtocol.Field header = default;
            while (true)
            {
                reader.ReadFieldHeader(ref header);
                if (header.IsEndBaseOrEndObject)
                    break;
                id += header.FieldIdDelta;
                if (id == 0)
                {
                    instance.HighlightColor = (global::Game.Core.ClickPipeHighlightModel)_codec1.ReadValue(ref reader, header);
                    reader.ReadFieldHeader(ref header);
                    if (header.IsEndBaseOrEndObject)
                        break;
                    id += header.FieldIdDelta;
                }

                if (id == 1)
                {
                    instance.Staged = (global::System.Collections.Generic.List<global::OneOf.OneOf<global::Game.Core.GameCardModel, global::Game.Core.GamePlayerModel>>)_codec0.ReadValue(ref reader, header);
                    reader.ReadFieldHeader(ref header);
                    if (header.IsEndBaseOrEndObject)
                        break;
                    id += header.FieldIdDelta;
                }

                if (id == 2)
                {
                    instance.CanStage = (global::System.Collections.Generic.List<global::OneOf.OneOf<global::Game.Core.GameCardModel, global::Game.Core.GamePlayerModel>>)_codec0.ReadValue(ref reader, header);
                    reader.ReadFieldHeader(ref header);
                    if (header.IsEndBaseOrEndObject)
                        break;
                    id += header.FieldIdDelta;
                }

                if (id == 3)
                {
                    instance.Text = (string)global::Orleans.Serialization.Codecs.StringCodec.ReadValue(ref reader, header);
                    reader.ReadFieldHeader(ref header);
                    if (header.IsEndBaseOrEndObject)
                        break;
                    id++;
                }

                reader.ConsumeUnknownField(ref header);
            }
        }
```

**Comment by luckyycode** [comment]
same here, any news about this issue?

**Comment by luckyycode** [comment]
<img width="780" alt="Screenshot 2023-08-25 at 16 58 36" src="https://github.com/dotnet/orleans/assets/6823752/402ea614-f336-4cdd-8042-cb3f373c5850">

also this does look very suspicious to me

**Comment by ReubenBond** [comment]
Thank you for your patience, @luckyycode & @zpxp. I have opened a PR to fix this: #8586
